frodo.js

```diff
@@ -1,44 +1,40 @@
-/**
```

Comment:
SCRIPT-This function might benefit from using async/await instead of callbacks for better readability.